File: C#/Xamarin Presentation/Classes/MainActivity.cs
using System;
using System.Collections.Generic;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Widget;
using AndroidX.Annotations;
using AndroidX.AppCompat.Widget;
using AndroidX.AppCompat.App;
using Java.Lang;
using Xamarin_Presentation.Classes;
using String = Java.Lang.String;

namespace Xamarin_Presentation
{
    [Activity(MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        ListView contactView;
        ContactList contactList = new ContactList();
        HashSet<int> editedValues = new HashSet<int>();
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);
            contactView = FindViewById(Resource.Id.button_list) as ListView;
            contactView.Adapter = new ContactsAdapter(this, contactList, editedValues);
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            base.OnActivityResult(requestCode, resultCode, data);

            if (data == null || requestCode != 3 || resultCode != Result.Ok) return;

            string firstName = data.GetStringExtra("FirstName");
            string lastName = data.GetStringExtra("LastName");
            int idx = data.GetIntExtra("Idx", -1);

            if (string.IsNullOrWhiteSpace(firstName) || string.IsNullOrWhiteSpace(lastName) || idx == -1) return;

            ContactInfo contact = contactList.Records[idx];
            contact.firstName = firstName;
            contact.lastName = lastName;

            editedValues.Add(idx);

            BaseAdapter adapter = (BaseAdapter)contactView.Adapter;
            adapter.NotifyDataSetChanged();
        }
    }
}

File: Java/app/src/main/java/com/example/xamarinpresentationjava/MainActivity.java
package com.example.xamarinpresentationjava;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.BaseAdapter;
import android.widget.ListView;

import java.util.HashSet;

public class MainActivity extends AppCompatActivity {
    ListView contactView;
    ContactList contactList = new ContactList();
    HashSet<Integer> editedValues = new HashSet<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        contactView = findViewById(R.id.button_list);
        contactView.setAdapter(new ContactsAdapter(this, contactList, editedValues));
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if(data == null || requestCode != 3 || resultCode != RESULT_OK) return;

        String firstName = data.getStringExtra("FirstName");
        String lastName = data.getStringExtra("LastName");
        int idx = data.getIntExtra("Idx", -1);

        if(firstName.equals("") || lastName.equals("") || idx == -1) return;

        ContactInfo contact = contactList.getRecords()[idx];
        contact.firstName = firstName;
        contact.lastName = lastName;

        editedValues.add(idx);

        BaseAdapter adapter = (BaseAdapter) contactView.getAdapter();
        adapter.notifyDataSetChanged();
    }
}